core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/ScheduleGenerator.scala

```diff
@@ -173,6 +170,21 @@ abstract class ScheduleGenerator(
       .addOperator(matWriterPhysicalOp)
       .addLink(sourceToWriterLink)
 
+    // expect exactly one input port and one output port
+    val schema = newPhysicalPlan
```

Comment:
can we just use the newly created `matWriterPhysicalOp` to get the schema?

Reply:
No, that copy doesn’t have propagated output schema. Scheme is only available after the link is added.